python_modules/libraries/dagster-airbyte/dagster_airbyte/resources.py

```diff
@@ -0,0 +1,205 @@
+import logging
+import time
+import json
+from typing import Optional
+
+import requests
+from dagster import (
+    get_dagster_logger, 
+    __version__,
+    resource,
+    StringSource,
+    Failure,
+    Field
+)
+from dagster_airbyte.types import AirbyteOutput
+from requests.exceptions import RequestException
+
+DEFAULT_POLL_INTERVAL_SECONDS = 10
+class AirbyteResource:
+    """
+    This class exposes methods on top of the Airbyte REST API.
+    """
+    RUNNING = "running"
+    SUCCEEDED = "succeeded"
+    CANCELLED = "cancelled"
+    PENDING = "pending"
+    FAILED = "failed"
+    ERROR = "error"
+    INCOMPLETE = "incomplete"
+
+    def __init__(
+        self,
+        host: str = "localhost",
+        port: str = "8000",
+        request_max_retries: int = 3,
+        request_retry_delay: float = 0.25,
+        log: logging.Logger = get_dagster_logger(),
+    ):
+        self._host = host
+        self._port = port
+        self._request_max_retries = request_max_retries
+        self._request_retry_delay = request_retry_delay
+
+        self._log = log
+
+    @property
+    def api_base_url(self) -> str:
+        return f"http://{self._host}:{self._port}/api/v1"
+    
+    def make_request(self, endpoint: str, data: Optional[str]):
```

Comment:
Is this compatible with Airbyte Cloud? I assume to make API calls against it, we'll need some way to provide an API token?

Reply:
At the moment the Airbyte Cloud API is not available. It's not clear to answer :(